components/camel-google-bigquery/src/main/java/org/apache/camel/component/google/bigquery/GoogleBigQueryProducer.java

```diff
@@ -0,0 +1,219 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.google.bigquery;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.api.client.util.Strings;
+import com.google.api.services.bigquery.Bigquery;
+import com.google.api.services.bigquery.model.TableDataInsertAllRequest;
+import com.google.api.services.bigquery.model.TableDataInsertAllResponse;
+import com.google.api.services.bigquery.model.TableRow;
+import org.apache.camel.AsyncCallback;
+import org.apache.camel.Exchange;
+import org.apache.camel.impl.DefaultAsyncProducer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Generic BigQuery Producer
+ */
+public class GoogleBigQueryProducer extends DefaultAsyncProducer {
+
+    private final Logger logger;
+    private final GoogleBigQueryConfiguration configuration;
+
+    public GoogleBigQueryProducer(GoogleBigQueryEndpoint endpoint, GoogleBigQueryConfiguration configuration) {
+        super(endpoint);
+
+        this.configuration = configuration;
+
+        String loggerId = configuration.getLoggerId();
+        if (loggerId == null || loggerId.trim().isEmpty()) {
+            loggerId = this.getClass().getName();
+        }
+
+        logger = LoggerFactory.getLogger(loggerId);
+    }
+
+    /**
+     * The method converts a single incoming message into a List
+     *
+     * @param exchange
+     * @return
+     */
+    private static List<Exchange> prepareExchangeList(Exchange exchange) {
+
+        List<Exchange> entryList;
+
+        if (null == exchange.getProperty(Exchange.GROUPED_EXCHANGE)) {
+            entryList = new ArrayList<>();
+            entryList.add(exchange);
+        } else {
+            entryList = (List<Exchange>) exchange.getProperty(Exchange.GROUPED_EXCHANGE);
+        }
+
+        return entryList;
+    }
+
+    /**
+     * Process the exchange
+     *
+     * The incoming exchange can be a grouped exchange in which case all the exchanges will be combined.
+     *
+     * The incoming can be
+     * <ul>
+     *     <li>A map where all map keys will map to field records. One map object maps to one bigquery row</li>
+     *     <li>A list of maps. Each entry in the list will map to one bigquery row</li>
+     * </ul>
+     * The incoming message is expected to be a List of Maps
+     * The assumptions:
+     * - All incoming records go into the same table
+     * - Incoming records sorted by the timestamp
+     */
+    @Override
+    public void process(Exchange exchange) throws Exception {
```

Comment:
What is the point with creating a list<exchanges> in a producer??? A producer should generally just store the incoming single exchange.